tests/Benchmarks/System.Text.JsonLab/JsonWriterPerf.cs

```diff
@@ -2,119 +2,142 @@
+            }
         }
 
         [Benchmark]
         public void WriterSystemTextJsonBasic()
         {
             _arrayFormatter.Clear();
-            WriterSystemTextJsonBasic(Formatted, _arrayFormatter);
+            if (IsUTF8Encoded)
+                WriterSystemTextJsonBasicUtf8(Formatted, _arrayFormatter, _data);
+            else
+                WriterSystemTextJsonBasicUtf16(Formatted, _arrayFormatter, _data);
         }
 
         [Benchmark]
         public void WriterNewtonsoftBasic()
         {
-            _stream.Seek(0, SeekOrigin.Begin);
-            WriterNewtonsoftBasic(Formatted, _writer);
+            TextWriter writer;
+            if (IsUTF8Encoded)
+            {
+                _memoryStream.Seek(0, SeekOrigin.Begin);
+                writer = _streamWriter;
+            }
+            else
+            {
+                _stringBuilder.Clear();
+                writer = new StringWriter(_stringBuilder);
+            }
+            WriterNewtonsoftBasic(Formatted, writer, _data);
         }
 
         [Benchmark]
         public void WriterSystemTextJsonHelloWorld()
         {
             _arrayFormatter.Clear();
-            WriterSystemTextJsonHelloWorld(Formatted, _arrayFormatter);
+            if (IsUTF8Encoded)
+                WriterSystemTextJsonHelloWorldUtf8(Formatted, _arrayFormatter);
+            else
+                WriterSystemTextJsonHelloWorldUtf16(Formatted, _arrayFormatter);
         }
 
         [Benchmark]
         public void WriterNewtonsoftHelloWorld()
         {
-            _stream.Seek(0, SeekOrigin.Begin);
-            WriterNewtonsoftHelloWorld(Formatted, _writer);
-        }
-
-        // Skipping this test for now since it results in too many extra permutations
-        // The Target Param is redundant here so we end up running this test twice.
-        //[Benchmark]
-        [ArgumentsSource(nameof(GetWriterSystemTextJsonParameters))]
-        public void WriterSlowSystemTextJsonHelloWorld(ArrayFormatter formatter)
-        {
-            formatter.Clear();
-            WriterSystemTextJsonHelloWorld(Formatted, formatter);
-        }
-
-        // Skipping this test for now since it results in too many extra permutations
-        // The Target Param is redundant here so we end up running this test twice.
-        //[Benchmark]
-        [ArgumentsSource(nameof(GetWriterSystemTextJsonParameters))]
-        public void WriterSlowSystemTextJsonBasic(ArrayFormatter formatter)
-        {
-            formatter.Clear();
-            WriterSystemTextJsonBasic(Formatted, formatter);
+            TextWriter writer;
+            if (IsUTF8Encoded)
+            {
+                _memoryStream.Seek(0, SeekOrigin.Begin);
+                writer = _streamWriter;
+            }
+            else
+            {
+                _stringBuilder.Clear();
+                writer = new StringWriter(_stringBuilder);
+            }
+            WriterNewtonsoftHelloWorld(Formatted, writer);
```

Comment:
Could you please move the duplicated code to a helper methd?

```cs
TextWriter writer;
if (IsUTF8Encoded)
{
	_memoryStream.Seek(0, SeekOrigin.Begin);
	writer = _streamWriter;
}
else
{
	_stringBuilder.Clear();
	writer = new StringWriter(_stringBuilder);
}
```

Reply:
Good call, will do.